lib/posix/pthread_barrier.c

```diff
@@ -160,6 +160,40 @@ int pthread_barrier_destroy(pthread_barrier_t *b)
 	return 0;
 }
 
+int pthread_barrierattr_init(pthread_barrierattr_t *attr)
+{
+	__ASSERT_NO_MSG(attr != NULL);
+
+	((pthread_barrierattr_t *) attr)->pshared = PTHREAD_PROCESS_DEFAULT;
```

Comment:
Is the cast needed?

```suggestion
	attr->pshared = PTHREAD_PROCESS_DEFAULT;
```

Reply:
I don't think so - commented on that previously as well

Reply:
Would it be necessary if the `pthread_barrierattr_t` struct is implementation defined?

Reply:
Zephyr is the implementation, and we are defining it 😁 

The only thing to maybe watch out for is that other libc's might already have this typed as an `uint32_t` or something. In that case, we probably don't even need to make a `struct` out of it.

I'm going to run a couple of tests on your PR locally and will report back some additional suggestions.

Reply:
> Zephyr is the implementation, and we are defining it

What about other libc's being linked though? Do they have their own implementations, or will they use this?

Reply:
They will use this.